spec/models/link_spec.rb

```diff
@@ -4,31 +4,98 @@
 
   let(:google) { Link.create(given_url: 'http://google.com') }
 
-  context 'Associations' do
-    it { should have_many :visits }
-    it { should belong_to :user }
+  describe 'associations' do
+    it { is_expected.to have_many :visits }
+    it { is_expected.to belong_to :user }
   end
 
-  context 'Validations' do
+  describe 'validations' do
     subject { Link.new(given_url: 'http://youtube.com') }
-    it { should validate_presence_of :given_url }
-    it { should validate_presence_of :slug }
-    it { should validate_uniqueness_of :slug }
-
-    it { should have_db_column :slug }
-    it { should have_db_column :title }
-    it { should have_db_column :clicks }
-    it { should have_db_column :user_id }
-    it { should have_db_index :slug }
-    it { should have_db_index :given_url }
+    it { is_expected.to validate_presence_of :given_url }
+    it { is_expected.to validate_uniqueness_of :slug }
+
+    it { is_expected.to have_db_column :slug }
+    it { is_expected.to have_db_column :title }
+    it { is_expected.to have_db_column :clicks }
+    it { is_expected.to have_db_column :user_id }
+    it { is_expected.to have_db_index :slug }
+    it { is_expected.to have_db_index :given_url }
   end
 
-  context 'Named Scopes' do
-    it 'should return the newest links first' do
-      expect(Link.newest_first.order_values).to eq ['links.created_at DESC']
+  describe 'instance methods' do
+    let(:link) { build(:link) }
+    subject { link }
+
+    it { is_expected.to respond_to :given_url }
+    it { is_expected.to respond_to :slug }
+    it { is_expected.to respond_to :title }
+    it { is_expected.to respond_to :clicks }
+
+    describe '#sanitize_slug' do
+      before do
+        link.slug = 'some slug'
+        link.sanitize_slug
+      end
+      it 'replaces spaces in slug with hyphen' do
+        expect(link.slug).to eq 'some-slug'
+      end
+    end
+
+    describe '#generate_slug' do
+      before { link.generate_slug }
+      it 'has a valid slug' do
+        expect(link.slug).to_not be_nil
+      end
+    end
+
+    describe '#display_slug' do
+      before { link.save }
+      it 'returns the full path with a slug' do
+        expect(link.display_slug).to eq ENV['BASE_URL'] + link.slug.to_s
+      end
+    end
+
+    describe '#add_visit_info' do
+      let(:visit) { create(:visit) }
+      before do
+        link.save
+        link.add_visit_info(visit)
+      end
+      it 'persists the visit info to the link' do
```

Comment:
**Nitpick - Code Readability**
Kindly add a newline before this line. Appropriately adding newlines enhances code readability.

Fix all instances of this issue.